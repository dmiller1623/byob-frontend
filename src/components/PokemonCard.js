import React, { Component } from 'react';
import Charizard from '../images/charizard.png'

class Pokemoncard extends Component {
  constructor(props) {
    super();
    this.state = {
    }
  };

  render () {
    const { name, type, img, speed, attack, defense, hp } = this.props;
    console.log(img)
    return(
      <div>
        <h1>{name}</h1>
        <img src={img}/>
      </div>
    )
  };
};

export default Pokemoncard;