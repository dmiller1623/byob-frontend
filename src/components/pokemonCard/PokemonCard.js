import React, { Component } from 'react';
import './pokemonCard.css'

class Pokemoncard extends Component {
  constructor(props) {
    super();
    this.state = {
      clicked: false
    }
  };
  
  handleClick = (e) => {
    e.preventDefault()
    let clicked = !this.state.clicked
    this.setState ({
      clicked
    })
  }

  render () {
    const { name, type, img, speed, attack, defense, hp, addToPokemonTeam } = this.props;

    return(
      <div 
        className='poke-card'>
        <h1>{name}</h1>
        <img onClick={(e) => this.handleClick(e)} src={img}/>
        {this.state.clicked && <p>Type: {type}</p>}
        {this.state.clicked && <p>Speed: {speed}</p>}
        {this.state.clicked && <p>Attack: {attack}</p>}
        {this.state.clicked && <p>Defense: {defense}</p>}
        {this.state.clicked && <p>HP: {hp}</p>}
        <button onClick={() => addToPokemonTeam(img)}>Add</button>
      </div>
    )
  };
};

export default Pokemoncard;