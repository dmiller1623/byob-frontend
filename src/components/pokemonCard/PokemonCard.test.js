import React from 'react';
import { shallow } from 'enzyme'; 
import PokemonCard from './PokemonCard.js';

describe('PokemonCard', () => {
  let wrapper;
 
  it('should match the snapshot', () => {
    wrapper = shallow(<PokemonCard />);
    expect(wrapper).toMatchSnapshot();
  });
});