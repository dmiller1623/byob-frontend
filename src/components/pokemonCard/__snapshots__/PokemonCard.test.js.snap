// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pokemoncard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="poke-card-name"
        />,
        <div>
          <img
            alt="left click button"
            className="poke-card-btn"
            onClick={[Function]}
            src="Add.png"
          />
          <img
            alt="right click button"
            className="poke-card-btn"
            onClick={[Function]}
            src="Stats.png"
          />
        </div>,
        <img
          alt="main pokemon"
          className="main-poke-img"
        />,
        false,
        false,
        false,
        false,
        false,
      ],
      "className": "poke-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "poke-card-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="left click button"
              className="poke-card-btn"
              onClick={[Function]}
              src="Add.png"
            />,
            <img
              alt="right click button"
              className="poke-card-btn"
              onClick={[Function]}
              src="Stats.png"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "left click button",
              "className": "poke-card-btn",
              "onClick": [Function],
              "src": "Add.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "right click button",
              "className": "poke-card-btn",
              "onClick": [Function],
              "src": "Stats.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "main pokemon",
          "className": "main-poke-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      false,
      false,
      false,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="poke-card-name"
          />,
          <div>
            <img
              alt="left click button"
              className="poke-card-btn"
              onClick={[Function]}
              src="Add.png"
            />
            <img
              alt="right click button"
              className="poke-card-btn"
              onClick={[Function]}
              src="Stats.png"
            />
          </div>,
          <img
            alt="main pokemon"
            className="main-poke-img"
          />,
          false,
          false,
          false,
          false,
          false,
        ],
        "className": "poke-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "poke-card-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="left click button"
                className="poke-card-btn"
                onClick={[Function]}
                src="Add.png"
              />,
              <img
                alt="right click button"
                className="poke-card-btn"
                onClick={[Function]}
                src="Stats.png"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left click button",
                "className": "poke-card-btn",
                "onClick": [Function],
                "src": "Add.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right click button",
                "className": "poke-card-btn",
                "onClick": [Function],
                "src": "Stats.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "main pokemon",
            "className": "main-poke-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
        false,
        false,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
